pkg/virt-controller/services/virtiofs.go

```diff
@@ -171,18 +159,44 @@ func generateContainerFromVolume(volume *v1.Volume, image string, resources k8sv
 		},
 	}
 
+	var (
+		cmd  []string
+		args []string
+	)
+	s := volume.VolumeSource
+	switch {
+	case s.PersistentVolumeClaim != nil, s.DataVolume != nil:
+		cmd = []string{"/usr/bin/virtiofs-placeholder"}
+		args = []string{
+			"--socket", virtiofs.VirtiofsPlaceholderSocket(volume.Name),
+		}
+		volumeMounts = append(volumeMounts, k8sv1.VolumeMount{
+			Name:      virtiofs.ExtraVolName,
+			MountPath: virtiofs.PlaceholderSocketDir,
+		})
+	case s.ConfigMap != nil, s.DownwardAPI != nil, s.Secret != nil, s.ServiceAccount != nil:
```

Comment:
I really prefer this knowledge to be in a single place, like using `IsConfigVolume()` from `pkg/storage/types/pvc.go`, same for the next case with ` IsPVCVolume()`

Reply:
ok, I can move it

Reply:
Done